src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDurationTypeConverter.cs

```diff
@@ -51,7 +51,7 @@ public static InputDurationType CreateDurationType(ref Utf8JsonReader reader, st
             crossLanguageDefinitionId = crossLanguageDefinitionId ?? throw new JsonException("Duration type must have crossLanguageDefinitionId");
             encode = encode ?? throw new JsonException("Duration type must have encode");
             wireType = wireType ?? throw new JsonException("Duration type must have wireType");
-
+            encode = encode == "duration-constant" ? "constant" : encode;
```

Comment:
consider to combine this to L52

Reply:
we should not do it in this way, it would be hard to track these magic values.

Deserialization should just simply do "deserialization" without modification unless we could prove it is absolutely necessary

Therefore here we should deserialize this as is, and if other parts in our generator is expecting `constant` instead of `duration-constant`, those parts should change accordingly to properly handle `duration-constant`.

Reply:
I try to add something like `TryReadEncoding`, but that cannot keep `throw new JsonException($"Encoding of Duration type {encode} is unknown.");`. So changed by adding a `CreateDurationKnownEncoding`

Reply:
In Azure plugin, we don't have the ability to manipulate deserialization.
So, encoding is a decorator we can get from the model. Can we deal with this after the deserialization with `InputModelType.Decorators`?

Reply:
Created issue here: https://github.com/microsoft/typespec/issues/6621